driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -405,9 +405,52 @@ private void incrementAuthErrorMetric() {
         }
     }
 
-    private UnsupportedProtocolVersionException unsupportedProtocolVersionException(ProtocolVersion triedVersion, ProtocolVersion serverProtocolVersion) {
-        logger.debug("Got unsupported protocol version error from {} for version {} server supports version {}", address, triedVersion, serverProtocolVersion);
-        return new UnsupportedProtocolVersionException(address, triedVersion, serverProtocolVersion);
+    private static final Pattern SUPPORTED_VERSIONS_PATTERN = Pattern.compile("supported versions are \\(([^)]+?)\\)");
```

Comment:
Since C* has been pretty inconsistent about the error messages it returns by release, what about just not parsing the error message at all and then tying the next version to try based on the protocol version that we previously attempted?   The problem we are experiencing at the moment is because we were using information provided by the server, so I think we should just not trust it unless we can get something in the protocol specification about how to handle this scenario.

In the general case, we could simply just retry at the attempted protocol version - 1.  The only downside of this is that it may take multiple attempts to find the right protocol version, but this is only done at initialization anyways, so it doesn't seem to have that much cost.  We could also add a method to `ProtocolVersion` for requesting the next version to try in the case where it's more complex than that, this is actually what the nodejs-driver [is proposing](https://github.com/datastax/nodejs-driver/blob/1ed4a37dc010b0e57e0ae55bfb4704d97e2d622d/lib/types/protocol-version.js#L113-L127), although it still keeps the optimization to use the protocol version in the error response if it is different (like we also do here).